integ-test/src/test/java/org/opensearch/sql/sql/NestedIT.java

```diff
@@ -366,4 +368,34 @@ public void test_nested_in_where_as_predicate_expression_with_relevance_query()
     assertEquals(1, result.getInt("total"));
     verifyDataRows(result, rows(10, "a"));
   }
+
+  @Test
+  public void nested_function_with_date_types_as_object_arrays_within_arrays_test() {
+    String query = "SELECT nested(address.moveInDate) FROM " + TEST_INDEX_NESTED_SIMPLE;
+    JSONObject result = executeJdbcRequest(query);
+
+    assertEquals(11, result.getInt("total"));
+    verifyDataRows(result,
+        rows(new JSONObject(Map.of("dateAndTime","1984-04-12 09:07:42"))),
+        rows(new JSONArray(
+            List.of(
+                Map.of("dateAndTime", "2023-05-03 08:07:42"),
+                Map.of("dateAndTime", "2001-11-11 04:07:44"))
+            )
+        ),
+        rows(new JSONObject(Map.of("dateAndTime", "1966-03-19 03:04:55"))),
+        rows(new JSONObject(Map.of("dateAndTime", "2011-06-01 01:01:42"))),
+        rows(new JSONObject(Map.of("dateAndTime", "1901-08-11 04:03:33"))),
+        rows(new JSONObject(Map.of("dateAndTime", "2023-05-03 08:07:42"))),
+        rows(new JSONObject(Map.of("dateAndTime", "2001-11-11 04:07:44"))),
+        rows(new JSONObject(Map.of("dateAndTime", "1977-07-13 09:04:41"))),
+        rows(new JSONObject(Map.of("dateAndTime", "1933-12-12 05:05:45"))),
+        rows(new JSONObject(Map.of("dateAndTime", "1909-06-17 01:04:21"))),
+        rows(new JSONArray(
```

Comment:
Would it be relevant to add something testing errors received when an invalid date is submitted, such as "2023-02-30"?